Searches/IterativeBinarySearch.java

```diff
@@ -0,0 +1,59 @@
+
+import java.util.Arrays;
+import java.util.Random;
+
+/**
+ * 
+ * @author Gabriele La Greca : https://github.com/thegabriele97
+ * 
+ */
+
+public final class IterativeBinarySearch {
+
+    /**
+     * This method implements an iterative version of binary search algorithm
+     * 
+     * @param a sorted array
+     * @param the key to search in array
+     * 
+     * @return the index of key in the array or -1 if not found
+     */
+    public static <T extends Comparable<T>> int BS(T[] array, T key) {
```

Comment:
Please respect naming conventions in Java. Methods begin with lower case and use camel case.